File: jdroid-android-firebase-remoteconfig/src/main/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigFetchFcmMessage.java
package com.jdroid.android.firebase.remoteconfig;

import com.google.firebase.messaging.RemoteMessage;
import com.jdroid.android.firebase.fcm.AbstractFcmMessage;
import com.jdroid.android.utils.SharedPreferencesHelper;

public class FirebaseRemoteConfigFetchFcmMessage extends AbstractFcmMessage {

	@Override
	public String getMessageKey() {
		return "firebaseRemoteConfigFetchFcmMessage";
	}

	@Override
	public void handle(RemoteMessage remoteMessage) {
		SharedPreferencesHelper.get().savePreferenceAsync(FirebaseRemoteConfigLoader.CONFIG_STALE, true);
		new FirebaseRemoteConfigFetchCommand().start();
	}
}

File: jdroid-android-firebase-remoteconfig/src/main/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigFetchFcmMessage.kt
package com.jdroid.android.firebase.remoteconfig

import com.google.firebase.messaging.RemoteMessage
import com.jdroid.android.firebase.fcm.AbstractFcmMessage
import com.jdroid.android.utils.SharedPreferencesHelper

class FirebaseRemoteConfigFetchFcmMessage : AbstractFcmMessage() {

    override fun getMessageKey(): String {
        return "firebaseRemoteConfigFetchFcmMessage"
    }

    override fun handle(remoteMessage: RemoteMessage) {
        SharedPreferencesHelper.get().savePreferenceAsync(FirebaseRemoteConfigLoader.CONFIG_STALE, true)
        FirebaseRemoteConfigFetchCommand().start()
    }
}
